### PR TITLE
Dev/borrowing list detail

### PR DESCRIPTION
**Pull request by BornToLivee**
its PR after hard conflicts resolution. Please watch this code meticulously

### CODE REVIEW

File: borrowing/views.py [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+        user = self.request.user
+        if user.is_staff:
+            return Borrowing.objects.select_related("book", "user")
+        return (Borrowing.objects.filter(user=user)
+                .select_related("book", "user"))
+
+
+class BorrowingListView(BaseBorrowingView, generics.ListAPIView):
+    serializer_class = BorrowingListReadSerializer
+    filter_backends = [DjangoFilterBackend]
```
Comment (lilarin) [inline borrowing/views.py]: Specifying filter_backends is unnecessary bc of settings.py configurarion

File: borrowing/urls.py [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 
-from borrowing.views import BorrowingCreateView, BorrowingListView
+from borrowing.views import (
+    BorrowingListView,
+    BorrowingDetailView,
+    BorrowingCreateView
+)
 
 urlpatterns = [
     path("", BorrowingListView.as_view(), name="borrowing-list"),
```
Comment (lilarin) [inline borrowing/urls.py]: Can we use one route, but chose different serializers based on self.action and get_serializer_class?

File: borrowing/views.py [hunk 3]
```diff
@@ -1,8 +1,35 @@
...
-from .serializers import BorrowingCreateSerializer, BorrowingSerializer
+from borrowing.serializers import (
+    BorrowingListReadSerializer,
+    BorrowingRetrieveReadSerializer,
+    BorrowingCreateSerializer
+)
+
+
+class BaseBorrowingView(generics.GenericAPIView):
+    permission_classes = [permissions.IsAuthenticated]
```
Comment (BohdanKuzik) [inline borrowing/views.py]: You can write just [IsAuthenticated]

**Review by BohdanKuzik — APPROVED** [review]
Good job!

**Review by lilarin — APPROVED** [review]

**Pull request merged**